### PR TITLE
Add "state" field in CONFIG_DB a toggle of the fabric port monitor feature

### PR DESCRIPTION
**Pull request by jfeng-arista**
fabric port monitoring feature. The command to set this is "config fabric port monitor state <enable/disable>"

#### What I did
Add "state" field in CONFIG_DB fabric_monitor table as a toggle of the fabric port monitor feature.
The command is "config fabric port monitor state <enable/disable>"

Also add a test for this. The sonic mgmt test will be added shortly.

### CODE REVIEW

**Comment by jfeng-arista** [comment]
test_config_int_ip_rem_xxx tests failure, not related to this change

**Comment by jfeng-arista** [comment]
test_config_int_ip_rem_xxx tests failure, do not look like related to this change

**Comment by jfeng-arista** [comment]
test_config_int_ip_rem[ip_route_for_int_ip]
assert 1 == 0   +1   -0

test_config_int_ip_rem_static[ip_route_for_int_ip]
AssertionError: assert 'Error: Cannot remove the last IP entry of interface Ethernet2. A static ip route is still bound to the RIF.' in ''  +  where '' = <Result KeyError('namespace')>.output

the failures do not look like related to this change

**Comment by jfeng-arista** [comment]
test_config_int_ip_rem[ip_route_for_int_ip]
assert 1 == 0   +1   -0
test_config_int_ip_rem_sub_intf[ip_route_for_int_ip]
assert 1 == 0   +1   -0

it failed with above tests, it does not look like related to this change.

**Review by arlakshm — COMMENTED** [review]
as comments

File: config/fabric.py [hunk 1]
```diff
@@ -157,6 +157,44 @@ def error_threshold(crccells, rxcells, namespace):
...
+    # Update entry
+    config_db.mod_entry("FABRIC_MONITOR", "FABRIC_MONITOR_DATA",
+        {'monState': state})
+
+#
+# 'config fabric port montior state <enable/disable>'
+#
+@monitor.command()
+@click.argument('state', metavar='<state>', required=True)
+@multi_asic_util.multi_asic_click_option_namespace
```
[the remark this answers is not in the document]
Comment (jfeng-arista) [reply config/fabric.py:184]: The only reason that I added here is for this to be the same place and same format  as other threshold configuration ( that we discussed in the design meeting, they also per asic ) , example: 
config fabric port monitor error threshold <num>
config fabric port monitor poll threshold <num>
Let me know if this is ok to stay with other configuration.

File: config/fabric.py [hunk 2]
```diff
@@ -157,6 +157,44 @@ def error_threshold(crccells, rxcells, namespace):
...
+def state(state, namespace):
+    """FABRIC PORT MONITOR STATE configuration tasks"""
+    ctx = click.get_current_context()
+
+    n_asics = multi_asic.get_num_asics()
+    if n_asics > 1 and namespace is None:
+        ns = multi_asic.get_all_namespaces()
+        ns_list = ns['front_ns'] + ns['back_ns'] + ns['fabric_ns']
+        for namespace in ns_list:
+            setFabricPortMonitorState(state, namespace)
```
[the remark this answers is not in the document]
Comment (jfeng-arista) [reply config/fabric.py:194]: done

File: config/fabric.py [hunk 3]
```diff
@@ -157,6 +157,44 @@ def error_threshold(crccells, rxcells, namespace):
...
         {'monErrThreshCrcCells': crccells, 'monErrThreshRxCells': rxcells})
 
+def setFabricPortMonitorState(state, namespace ):
+    """ set the fabric port monitor state"""
+    # Connect to config database
+    config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+    config_db.connect()
+
+    # Connect to state database
+    state_db = SonicV2Connector(use_unix_socket_path=True, namespace=namespace)
```
[the remark this answers is not in the document]
Comment (jfeng-arista) [reply config/fabric.py:167]: done

**Review by arlakshm — DISMISSED** [review]

**Comment by judyjoseph** [comment]
@arlakshm could you  review again

**Comment by judyjoseph** [comment]
@vmittal-msft @saksarav-nokia

**Review by arlakshm — APPROVED** [review]

**Review by vmittal-msft — APPROVED** [review]

**Pull request merged**